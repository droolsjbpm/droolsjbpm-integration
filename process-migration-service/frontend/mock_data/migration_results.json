[
    {
      "id": 1,
      "migrationId": 1,
      "processInstanceId": null,
      "startDate": "2019-02-03T21:17:54.739Z",
      "endDate": "2019-02-03T21:17:54.854Z",
      "successful": true,
      "logs": [
        "INFO Sun Feb 03 16:17:54 EST 2019 Variable instances updated = 3 for process instance id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Node instances updated = 3 for process instance id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Process instances updated = 1 for process instance id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Task variables updated = 4 for process instance id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Task audit updated = 1 for process instance id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Tasks updated = 1 for process instance id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Context info updated = 0 for process instance id 1",
        "WARN Sun Feb 03 16:17:54 EST 2019 Source and target process id is exactly the same (evaluation) it's recommended to use unique process ids",
        "INFO Sun Feb 03 16:17:54 EST 2019 Mapping: Node instance logs to be updated  = [1]",
        "INFO Sun Feb 03 16:17:54 EST 2019 Mapping: Node instance logs updated = 1 for node instance id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Mapping: Task audit updated = 1 for task id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Mapping: Task updated = 1 for task id 1",
        "INFO Sun Feb 03 16:17:54 EST 2019 Migration of process instance (1) completed successfully to process evaluation"
      ]
    }
  ]