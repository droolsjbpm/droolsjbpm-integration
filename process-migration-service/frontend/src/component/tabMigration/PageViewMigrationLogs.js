import React, { Component } from "react";
import PropTypes from "prop-types";
import {
  sortableHeaderCellFormatter,
  actionHeaderCellFormatter,
  Icon,
  OverlayTrigger,
  Tooltip,
  Table,
  Wizard
} from "patternfly-react";

import { MigrationLog } from "./MigrationLog";
import AdvancedTable from "../common/AdvancedTable";

const tooltipViewLog = (
  <Tooltip id="tooltip">
    <div>View Log</div>
  </Tooltip>
);

const PrimaryLogDetailsContent = props => {
  const { log } = props;

  if (log != null && log != "") {
    return (
      <Table.Actions key="1">
        <MigrationLog log={log} />
      </Table.Actions>
    );
  } else {
    return <div> No detail migration log</div>;
  }
};

const emptyValueTableCellFormatter = value =>
  value ? <Table.Cell>{value}</Table.Cell> : <Table.Cell>N/A</Table.Cell>;

export default class PageViewMigrationLogs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLogDetailsShown: false,
      columns: [
        {
          property: "id",
          header: {
            label: "ID",
            props: {
              index: 0,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [],
            formatters: [],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 0
            },
            formatters: [emptyValueTableCellFormatter]
          }
        },
        {
          property: "migrationId",
          header: {
            label: "Migration ID",
            props: {
              index: 1,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [],
            formatters: [],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 1
            },
            formatters: [emptyValueTableCellFormatter]
          }
        },
        {
          property: "processInstanceId",
          header: {
            label: "Process Instance ID",
            props: {
              index: 2,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [],
            formatters: [],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 2
            },
            formatters: [emptyValueTableCellFormatter]
          }
        },
        {
          property: "startDate",
          header: {
            label: "Started Time",
            props: {
              index: 3,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [],
            formatters: [],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 3
            },
            formatters: [emptyValueTableCellFormatter]
          }
        },
        {
          property: "endDate",
          header: {
            label: "Finished Time",
            props: {
              index: 4,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [],
            formatters: [],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 4
            },
            formatters: [emptyValueTableCellFormatter]
          }
        },
        {
          property: "successful",
          header: {
            label: "Status",
            props: {
              index: 5,
              rowSpan: 1,
              colSpan: 1
            },
            transforms: [],
            formatters: [],
            customFormatters: [sortableHeaderCellFormatter]
          },
          cell: {
            props: {
              index: 5
            },
            formatters: [
              (value, { rowData }) => {
                if (rowData.successful) {
                  return (
                    <Table.Cell
                      key={
                        rowData.migrationId + "_" + rowData.id + "_" + "SUCCESS"
                      }
                      className="success"
                    >
                      SUCCESS
                    </Table.Cell>
                  );
                } else {
                  return (
                    <Table.Cell
                      key={
                        rowData.migrationId + "_" + rowData.id + "_" + "FAILED"
                      }
                      className="danger"
                    >
                      FAILED
                    </Table.Cell>
                  );
                }
              }
            ]
          }
        },
        {
          header: {
            label: "Actions",
            props: {
              index: 6,
              rowSpan: 1,
              colSpan: 1
            },
            formatters: [actionHeaderCellFormatter]
          },
          cell: {
            formatters: [
              (value, { rowData }) => [
                <Table.Actions key="0">
                  <OverlayTrigger overlay={tooltipViewLog} placement={"bottom"}>
                    <Table.Button
                      bsStyle="link"
                      onClick={() => this.showLogDetails(rowData)}
                    >
                      <Icon type="fa" name="info-circle" />
                    </Table.Button>
                  </OverlayTrigger>
                </Table.Actions>
              ]
            ]
          },
          property: "logs"
        }
      ],
      currentLogData: {}
    };
  }

  showLogDetails = log => {
    this.setState({ isLogDetailsShown: true, currentLogData: log });
  };

  hideLogDetails = () => {
    this.setState({ isLogDetailsShown: false });
  };

  render() {
    const { migrationLogs } = this.props;
    const { isLogDetailsShown, columns, currentLogData } = this.state;

    if (migrationLogs != null && migrationLogs != "") {
      return (
        <div>
          <Wizard
            show={isLogDetailsShown}
            onHide={this.hideLogDetails}
            id="logDetailWizard"
          >
            <Wizard.Header
              onClose={this.hideLogDetails}
              title="Migration Log Detail"
            />
            <Wizard.Body>
              <Wizard.Row>
                <Wizard.Main>
                  <Wizard.Contents stepIndex={0} activeStepIndex={0}>
                    <PrimaryLogDetailsContent log={currentLogData} />
                  </Wizard.Contents>
                </Wizard.Main>
              </Wizard.Row>
            </Wizard.Body>
          </Wizard>
          <AdvancedTable rows={migrationLogs} columns={columns} isSortable />
        </div>
      );
    } else {
      return <div> No detail migration logs for this status</div>;
    }
  }
}

PageViewMigrationLogs.propTypes = {
  migrationLogs: PropTypes.arrayOf(PropTypes.object)
};
