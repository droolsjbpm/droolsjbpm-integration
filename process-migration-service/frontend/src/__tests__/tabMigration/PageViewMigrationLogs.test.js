import renderer from "react-test-renderer";
import React from "react";
import PageViewMigrationLogs from "../../component/tabMigration/PageViewMigrationLogs";
import migration_results from "../../../mock_data/migration_results.json";

test("PageViewMigrationLogs renders correctly using snapshot", () => {
  const tree = renderer
    .create(<PageViewMigrationLogs migrationLogs={migration_results} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
