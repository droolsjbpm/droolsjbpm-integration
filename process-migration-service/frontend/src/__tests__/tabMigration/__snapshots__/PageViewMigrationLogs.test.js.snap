// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageViewMigrationLogs renders correctly using snapshot 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <table
      className="table dataTable table-striped table-bordered table-hover"
      role="grid"
    >
      <thead>
        <tr>
          <th
            aria-label="ID"
            className="sorting sort sort-none"
            colSpan={1}
            index={0}
            onClick={[Function]}
            rowSpan={1}
          >
            ID
          </th>
          <th
            aria-label="Migration ID"
            className="sorting sort sort-none"
            colSpan={1}
            index={1}
            onClick={[Function]}
            rowSpan={1}
          >
            Migration ID
          </th>
          <th
            aria-label="Process Instance ID"
            className="sorting sort sort-none"
            colSpan={1}
            index={2}
            onClick={[Function]}
            rowSpan={1}
          >
            Process Instance ID
          </th>
          <th
            aria-label="Started Time"
            className="sorting sort sort-none"
            colSpan={1}
            index={3}
            onClick={[Function]}
            rowSpan={1}
          >
            Started Time
          </th>
          <th
            aria-label="Finished Time"
            className="sorting sort sort-none"
            colSpan={1}
            index={4}
            onClick={[Function]}
            rowSpan={1}
          >
            Finished Time
          </th>
          <th
            aria-label="Status"
            className="sorting sort sort-none"
            colSpan={1}
            index={5}
            onClick={[Function]}
            rowSpan={1}
          >
            Status
          </th>
          <th
            aria-label="Actions"
            className=""
            colSpan={1}
            index={6}
            rowSpan={1}
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className=""
          >
            1
          </td>
          <td
            className=""
          >
            1
          </td>
          <td
            className=""
          >
            N/A
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.739Z
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.854Z
          </td>
          <td
            className="success"
          >
            SUCCESS
          </td>
          <td
            className="table-view-pf-actions"
          >
            <div
              className="table-view-pf-btn"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-info-circle"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            2
          </td>
          <td
            className=""
          >
            1
          </td>
          <td
            className=""
          >
            N/A
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.739Z
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.854Z
          </td>
          <td
            className="success"
          >
            SUCCESS
          </td>
          <td
            className="table-view-pf-actions"
          >
            <div
              className="table-view-pf-btn"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-info-circle"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            3
          </td>
          <td
            className=""
          >
            1
          </td>
          <td
            className=""
          >
            N/A
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.739Z
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.854Z
          </td>
          <td
            className="success"
          >
            SUCCESS
          </td>
          <td
            className="table-view-pf-actions"
          >
            <div
              className="table-view-pf-btn"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-info-circle"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            4
          </td>
          <td
            className=""
          >
            1
          </td>
          <td
            className=""
          >
            N/A
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.739Z
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.854Z
          </td>
          <td
            className="success"
          >
            SUCCESS
          </td>
          <td
            className="table-view-pf-actions"
          >
            <div
              className="table-view-pf-btn"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-info-circle"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            5
          </td>
          <td
            className=""
          >
            1
          </td>
          <td
            className=""
          >
            N/A
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.739Z
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.854Z
          </td>
          <td
            className="success"
          >
            SUCCESS
          </td>
          <td
            className="table-view-pf-actions"
          >
            <div
              className="table-view-pf-btn"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-info-circle"
                />
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            6
          </td>
          <td
            className=""
          >
            1
          </td>
          <td
            className=""
          >
            N/A
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.739Z
          </td>
          <td
            className=""
          >
            2019-02-03T21:17:54.854Z
          </td>
          <td
            className="success"
          >
            SUCCESS
          </td>
          <td
            className="table-view-pf-actions"
          >
            <div
              className="table-view-pf-btn"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <button
                className="btn btn-link"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-info-circle"
                />
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <form
      className="content-view-pf-pagination migration-log-table-pagination table-view-pf-pagination clearfix"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <div
          className="dropup pagination-pf-pagesize btn-group"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-default"
            disabled={false}
            id="pagination-row-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button"
          >
            6
             
            <span
              className="caret"
            />
          </button>
          <ul
            aria-labelledby="pagination-row-dropdown"
            className="dropdown-menu"
            role="menu"
          >
            <li
              className="active"
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                6
              </a>
            </li>
            <li
              className=""
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                10
              </a>
            </li>
            <li
              className=""
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                15
              </a>
            </li>
          </ul>
        </div>
        <span>
          per page
        </span>
      </div>
      <div
        className="form-group"
      >
        <span>
          <span
            className="pagination-pf-items-current"
          >
            1
            -
            6
          </span>
           
          of
           
          <span
            className="pagination-pf-items-total"
          >
            18
          </span>
        </span>
        <ul
          className="pagination pagination-pf-back"
        >
          <li
            className="disabled"
          >
            <a
              href="#"
              onClick={[Function]}
              title="First Page"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-double-left i"
              />
            </a>
          </li>
          <li
            className="disabled"
          >
            <a
              href="#"
              onClick={[Function]}
              title="Previous Page"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-left i"
              />
            </a>
          </li>
        </ul>
        <label
          className="sr-only control-label"
        >
          Current Page
        </label>
        <input
          className="pagination-pf-page form-control"
          onChange={[Function]}
          type="text"
          value={1}
        />
        <span>
           
          of
           
          <span
            className="pagination-pf-pages"
          >
            3
          </span>
        </span>
        <ul
          className="pagination pagination-pf-forward"
        >
          <li
            className=""
          >
            <a
              href="#"
              onClick={[Function]}
              title="Next Page"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-right i"
              />
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="#"
              onClick={[Function]}
              title="Last Page"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-double-right i"
              />
            </a>
          </li>
        </ul>
      </div>
    </form>
  </div>
</div>
`;
